src/core/Debug.ts

```diff
@@ -0,0 +1,121 @@
+import {
+  Disposable,
+  ExtensionMode,
+  Location,
+  TextEditorSelectionChangeEvent,
+  window,
+  workspace,
+} from "vscode";
+import { SyntaxNode, TreeCursor } from "web-tree-sitter";
+import { Graph } from "../typings/Types";
+
+const originalDebugLog = console.debug;
+const disabledDebugLog = () => {};
+
+console.debug = disabledDebugLog;
+
+export default class Debug {
+  getNodeAtLocation?: any;
+  disposable?: Disposable;
+
+  constructor(private graph: Graph) {
+    this.graph.extensionContext.subscriptions.push(this);
+
+    this.evaluateSetting = this.evaluateSetting.bind(this);
+
+    switch (this.graph.extensionContext.extensionMode) {
+      // Development mode. Always enable.
+      case ExtensionMode.Development:
+        this.enableDebugLog();
+        break;
+      // Test mode. Always disable.
+      case ExtensionMode.Test:
+        this.disableDebugLog();
+        break;
+      // Production mode. Enable based on user setting.
+      case ExtensionMode.Production:
+        this.evaluateSetting();
+        workspace.onDidChangeConfiguration(this.evaluateSetting);
+        break;
+    }
+  }
+
+  init(getNodeAtLocation: any) {
```

Comment:
I'd prefer if this one came from the graph like we do with command server api, but not a showstopper

Reply:
Fixed that